### ISSUE TITLE
Multiple ISF schedule

### ISSUE THREAD

**Issue by pampopcorn** [openaps/oref0#567]
Having the same ISF that starts and ends at multiple times (like midnight:80, 6am:80, 10pm:80) is causing issues with autosense

**Comment by scottleibrand** [comment]
Per https://gitter.im/nightscout/intend-to-bolus?at=5976a581f5b3458e307c4615:

```
cat autotune/profile.json | jq . | grep sens
"autosens_max": 1.2,
"autosens_min": 0.7,
"rewind_resets_autosens": false,
"autosens_adjust_targets": true,
"sens": 113.977,
"sensitivities": [
"sensitivity": 113.977,
"sensitivity": 60,
"sensitivity": 60,

cat autotune/profile.pump.json | jq . | grep sens
"autosens_max": 1.2,
"autosens_min": 0.7,
"rewind_resets_autosens": true,
"autosens_adjust_targets": true,
"sens": 80,
"sensitivities": [
"sensitivity": 80,
"sensitivity": 80,
"sensitivity": 80,
```

It appears that autotune is populating profile.json with multiple ISF schedules if they exist.  It never tunes them (since they're not supported), but it leaves them there and they apparently are getting used instead of the autotuned one (at the times of day they're set for): `Autosens adjusting basal from 0.7 to 0.8; target_bg from 102.5 to 97; sens from 60 to 51.7 (autosens ratio 1.16)`

**Comment by scottleibrand** [comment]
I believe @tim2000s has also reported this same issue, but we never really dug into it with him.

**Comment by JELCRAWFORD** [comment]
I think we may be seeing the same issue as @pampopcorn, with lowtemping for long periods of time after bolus.  Have changed to a single ISF to see if that is a work around.

**Comment by JELCRAWFORD** [comment]
As an initial assessment (single bolus only) With multiple ISF's reduced to one from three, prediction lines seem to be back to 'normal' and the rig is not low temping they boy at this point.  Remains to be seen how 20% variance that autosense/autotune can manage evening-time high insulin needs (which are greater than 20% from average).  Is this restriction easily modifiable?

**Comment by pampopcorn** [comment]
After reducing ISF to just one schedule, the low temp is for less time.  It still gave me a zero temp about 20 minutes after bolus, but only for about 30-40 minutes.  Will report if it has any issues.

**Comment by JELCRAWFORD** [comment]
yes, I may have spoken too soon - it appears that we have a low temp running now too.  Will also update as we see how that goes.  Prediction lines do seem back to normal...

**Comment by pampopcorn** [comment]
I also might have spoken too soon, its back to zero for me too

**Comment by scottleibrand** [comment]
Zero temps should still be expected any time eventualBG (with carbs) remains below target, or if carb absorption remains slow after snoozeBG drops below target. Removing the multiple ISF schedules should just fix the predictions so they're accurate based on what it knows at the time.

**Comment by scottleibrand** [comment]
What additional action is required here?  Do we need to update the docs to note that multiple ISFs are not supported, and suggest people change their pump to use a single ISF?

**Comment by JELCRAWFORD** [comment]
Yes - the behavior from above seems to have been resolved with the single ISF. I agree that it should be documented that multiple ISF's are currently not supported. I'm not sure how others approached the single ISF conversion - we used an average ISF to try to accommodate the large early nighttime difference, at least to some degree, versus simply using the autotune suggested.  Tell me where you want it in the docs and I'll take a stab at the pr...
